libevmasm/Assembly.cpp

```diff
@@ -186,11 +186,11 @@ class Functionalizer
 
 	void printLocation()
 	{
-		if (!m_location.sourceName && m_location.isEmpty())
+		if (!m_location.source && m_location.isEmpty())
 			return;
 		m_out << m_prefix << "    /*";
-		if (m_location.sourceName)
-			m_out << " \"" + *m_location.sourceName + "\"";
+		if (m_location.source->name() != "")
```

Comment:
I think the empty source name should not receive any special treatment. That is what we have the empty source pointer for.